pkgs/build-support/rust/lib/default.nix

```diff
@@ -3,12 +3,14 @@
 rec {
   # https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch
   toTargetArch = platform:
-    if platform.isAarch32 then "arm"
+    /**/ if platform ? rustc.platform then platform.rustc.platform.arch
```

Comment:
`platform.rustc.platform` would be a rust-specific description of the platform, the target json converted to a Nix value, which must contain an `arch` field.